scheduler/scripts/run_main.py

```diff
@@ -104,7 +108,9 @@ def main(*,
     overall_plans: Dict[NightIndex, Plans] = {}
     nightly_timeline = NightlyTimeline()
 
-    for night_idx in sorted(night_indices):
+    curr = start.to_datetime()
+    night_idx = NightIndex(0)
+    while curr <= end.to_datetime():
```

Comment:
This seems like a common pattern. Is there any way we can extract it and pass in a function with the necessary details to perform each step?

Reply:
There is a lot of duplicate code between the scripts and Service.